pkg/addon/controllers/addontemplate/controller.go

```diff
@@ -88,27 +97,98 @@ func NewAddonTemplateController(
 		// easy to mock in unit tests
 		c.runControllerFunc = c.runController
 	}
-	return factory.New().WithInformersQueueKeysFunc(
-		queue.QueueKeyByMetaNamespaceName,
-		addonInformers.Addon().V1alpha1().ClusterManagementAddOns().Informer()).
+
+	controller := factory.New().
+		WithSyncContext(syncCtx).
+		WithInformersQueueKeysFunc(
+			queue.QueueKeyByMetaNamespaceName,
+			addonInformers.Addon().V1alpha1().ClusterManagementAddOns().Informer()).
 		WithBareInformers(
+			addonInformers.Addon().V1alpha1().ManagedClusterAddOns().Informer(),
 			// do not need to queue, just make sure the controller reconciles after the addonTemplate cache is synced
 			// otherwise, there will be "xx-addon-template" not found" errors in the log as the controller uses the
 			// addonTemplate lister to get the template object
 			addonInformers.Addon().V1alpha1().AddOnTemplates().Informer()).
 		WithSync(c.sync).
 		ToController("addon-template-controller", recorder)
+
+	// Add custom event handler for ManagedClusterAddon to filter configReference changes
+	_, err := addonInformers.Addon().V1alpha1().ManagedClusterAddOns().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			// Only handle configReference updates for AddOnTemplates
+			oldMCA, ok := oldObj.(*addonv1alpha1.ManagedClusterAddOn)
+			if !ok {
+				return
+			}
+			newMCA, ok := newObj.(*addonv1alpha1.ManagedClusterAddOn)
+			if !ok {
+				return
+			}
+
+			// Extract AddOnTemplate configReferences from both old and new
+			oldTemplateRefs := extractAddOnTemplateConfigRefs(oldMCA)
```

Comment:
can we just filter the mca based on whether it is an addonTemplate based addon?

Reply:
This will not only filter the addonTemplate-based addon, but also filter only the template that has changes.
Do you mean we do not need to compare whether the template has changed, we enqueue all update events as long as it is template-based?

Reply:
I think so, why do we only compare template change? If this is only for deleting, should we just filter the one that is template addon and is in deleting?